### PR TITLE
Documentation Refactoring

### PR DESCRIPTION
**Pull request by ghost**
This PR introduces the website-src directory which contains go templates and a script to build and compose these templates into the full markdown documentation in website.

It also includes new documentation for Compute services.

### CODE REVIEW

File: website_src/docs/r/arguments/package.markdown.tmpl [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{{define "package"}}* `package` - (Optional) A Wasm deployment package to upload. Defined below.{{end}}
+
+{{define "package_block"}}
+The `package` block supports uploading or modifying Wasm packages for use in a Fastly compute service. See Fastly's documentation on
```
Comment (phamann) [inline website_src/docs/r/arguments/package.markdown.tmpl]:
```suggestion
The `package` block supports uploading or modifying Wasm packages for use in a Fastly Compute@Edge service. See Fastly's documentation on
```

File: website_src/docs/r/arguments/package.markdown.tmpl [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+[Compute@Edge](https://www.fastly.com/products/edge-compute/serverless)
+
+* `filename` - (Required) The path to the Wasm deployment package within your local filesystem.
+* `source_code_hash` - (Optional) Used to trigger updates. Must be set to a SHA512 hash of the package file specified with the filename. The usual way to set this is filesha512("file.zip") (Terraform 0.11.12 and later) or filesha512(file("file.zip")) (Terraform 0.11.11 and earlier), where "file.zip" is the local filename of the Wasm deployment package.
```
Comment (phamann) [inline website_src/docs/r/arguments/package.markdown.tmpl]:
```suggestion
* `source_code_hash` - (Optional) Used to trigger updates. Must be set to a SHA512 hash of the package file specified with the filename. The usual way to set this is filesha512("package.tar.gz") (Terraform 0.11.12 and later) or filesha512(file("package.tar.gz")) (Terraform 0.11.11 and earlier), where "package.tar.gz" is the local filename of the Wasm deployment package.
```

File: website_src/docs/r/service_compute.html.markdown.tmpl [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+sidebar_current: "docs-fastly-resource-service-compute"
+description: |-
+  Provides an Fastly Compute Service
+---
+
+# fastly_service_compute
+
+Provides a Fastly Compute Service, representing the Wasm edge compute package which manages the configuration for a website, app,
+API, or anything else to be served through Fastly at the edge. A Service encompasses Domains
+and Backends.
```
Comment (phamann) [inline website_src/docs/r/service_compute.html.markdown.tmpl]:
```suggestion
Provides a Fastly Compute@Edge service. Compute@Edge is a computation platform capable of running custom binaries that you compile on your own systems and upload to Fastly. Security and portability is provided by compiling your code to [WebAssembly](https://webassembly.org/), which is ran at the edge using [Lucet](https://github.com/bytecodealliance/lucet), an open-source WebAssembly runtime created by Fastly. A compute service encompasses Domains and Backends.
```

File: website_src/docs/r/service_compute.html.markdown.tmpl [hunk 4]
```diff
@@ -0,0 +1,118 @@
+{{define "service_compute"}}---
+layout: "fastly"
+page_title: "Fastly: service_compute"
+sidebar_current: "docs-fastly-resource-service-compute"
+description: |-
+  Provides an Fastly Compute Service
```
Comment (phamann) [inline website_src/docs/r/service_compute.html.markdown.tmpl]:
```suggestion
  Provides an Fastly Compute@Edge service
```

File: website_src/docs/r/arguments/package.markdown.tmpl [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{{define "package"}}* `package` - (Optional) A Wasm deployment package to upload. Defined below.{{end}}
+
+{{define "package_block"}}
+The `package` block supports uploading or modifying Wasm packages for use in a Fastly compute service. See Fastly's documentation on
+[Compute@Edge](https://www.fastly.com/products/edge-compute/serverless)
```
Comment (phamann) [inline website_src/docs/r/arguments/package.markdown.tmpl:5]: @triblondon what is the best resource/URL to use here?

**Review by phamann — CHANGES_REQUESTED** [review]
This is looking great @guyrichardson. So nice to separate all of the blocks into their own partials. I've just some content change requests for the compute service.

File: website_src/docs/r/service_compute.html.markdown.tmpl (hunk 4 above)
Comment (ghost) [reply website_src/docs/r/service_compute.html.markdown.tmpl]: Done

File: website_src/docs/r/service_compute.html.markdown.tmpl (hunk 3 above)
Comment (ghost) [reply website_src/docs/r/service_compute.html.markdown.tmpl]: Done

File: website_src/docs/r/arguments/package.markdown.tmpl (hunk 2 above)
Comment (ghost) [reply website_src/docs/r/arguments/package.markdown.tmpl]: Done

File: website_src/docs/r/arguments/package.markdown.tmpl (hunk 1 above)
Comment (ghost) [reply website_src/docs/r/arguments/package.markdown.tmpl]: Done

**Review by phamann — APPROVED** [review]
LGTM

**Pull request merged**